docs/source/advanced/apollo-ast.mdx

```diff
@@ -42,24 +42,15 @@ val graphQLText = """
     }
 """.trimIndent()
 
-val parseResult = graphQLText.parseAsGQLDocument()
+val parseResult = Buffer().writeUtf8(graphQLText).parseAsGQLDocument()
 ```
 
-This method returns a `ParseResult<GQLDocument>`, which is a sealed class that can either be a `Success` or an `Error`:
+This method returns a `GQLResult<GQLDocument>`, which contains the document and/or parsing issues, each of which can have a severity of either `WARNING` or `ERROR`. Because there can be warnings, it is possible to have both a valid document and issues at the same time.
 
-```kotlin
-val queryGqlDocument: GQLDocument = when (parseResult) {
-  is ParseResult.Success -> parseResult.value
-  is ParseResult.Error -> throw MyException(parseResult.issues)
-}
-```
-
-If parse errors are encountered, their details are included in `issues`.
-
-If you don't expect errors, the `toGQLDocument` convenience method returns the AST directly:
+To get the document and throw on errors, use `valueAssertNoErrors()`:
 
 ```kotlin
-val queryGqlDocument: GQLDocument = graphQLText.toGQLDocument()
+val queryGqlDocument = parseResult.valueAssertNoErrors()
```

Comment:
I realized writing this that it is not consistent with `val ApolloResponse.dataAssertNoErrors` (val vs fun). Will most likely tweak that again for rc since it's not a very widespread API